widget/widget.go

```diff
@@ -19,9 +18,8 @@ type BaseWidget struct {
 	position async.Position
 	Hidden   bool
 
-	impl         atomic.Pointer[fyne.Widget]
-	propertyLock sync.RWMutex
-	themeCache   fyne.Theme
+	impl       atomic.Pointer[fyne.Widget]
```

Comment:
Should atomic types and things from the `async` package also be removed or is that perhaps better to have as a separate PR?

Reply:
I was wondering the same. I think later PR as it could impact semantics too